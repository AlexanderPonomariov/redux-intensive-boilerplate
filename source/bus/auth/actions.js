// Types
import { types } from './types';

export const authActions = Object.freeze({
    initialize: () => ({
        type: types.INITIALIZE,
    }),
    authenticate: () => ({
        type: types.AUTHENTICATE,
    }),
    logout: () => ({
        type: types.LOGOUT,
    }),
});
