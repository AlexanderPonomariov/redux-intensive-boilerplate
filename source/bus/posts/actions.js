// Types
import { types } from './types';

export const postsActions = Object.freeze({
    fillPosts: (posts) => ({
        type:    types.FILL_POSTS,
        payload: posts,
    }),
    createPost: (post) => ({
        type:    types.CREATE_POST,
        payload: post,
    }),
    removePost: (postId) => ({
        type:    types.REMOVE_POST,
        payload: postId,
    }),
    likePost: (likedPostIDs) => ({
        type:    types.LIKE_POST,
        payload: likedPostIDs,
    }),
    unlikePost: (unlikedPostIDs) => ({
        type:    types.UNLIKE_POST,
        payload: unlikedPostIDs,
    }),
    clearPosts: () => ({
        type: types.CLEAR_POSTS,
    }),
});
