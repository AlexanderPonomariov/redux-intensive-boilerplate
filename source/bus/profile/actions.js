// Types
import { types } from './types';

export const profileActions = Object.freeze({
    fillProfile: (user) => ({
        type:    types.FILL_PROFILE,
        payload: user,
    }),
    updateAvatar: (avatarUrl) => ({
        type:    types.UPDATE_AVATAR,
        payload: avatarUrl,
    }),
    clearProfile: () => ({
        type: types.CLEAR_PROFILE,
    }),
});
