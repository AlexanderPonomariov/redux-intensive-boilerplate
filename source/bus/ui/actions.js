// Instruments
import { types } from './types';

export const uiActions = Object.freeze({
    emitError: (error, meta = null) => ({
        type:    types.EMIT_ERROR,
        payload: error,
        error:   true,
        meta,
    }),
    initialize: () => ({
        type: types.INITIALIZE,
    }),
    initializeSuccess: () => ({
        type: types.INITIALIZE_SUCCESS,
    }),
    setPostsFetchingState: (postsFetchingState) => ({
        type:    types.SET_POSTS_FETCHING_STATE,
        payload: postsFetchingState,
    }),
    setAuthFetchingState: (authFetchingState) => ({
        type:    types.SET_AUTH_FETCHING_STATE,
        payload: authFetchingState,
    }),
    setOnlineState: (onlineState) => ({
        type:    types.SET_ONLINE_STATE,
        payload: onlineState,
    }),
    setProfileFetchingState: (state) => ({
        type:    types.SET_PROFILE_FETCHING_STATE,
        payload: state,
    }),
    setProfileEditingState: (state) => ({
        type:    types.SET_PROFILE_EDITING_STATE,
        payload: state,
    }),
    setAvatarFetchingState: (state) => ({
        type:    types.SET_AVATAR_EDITING_STATE,
        payload: state,
    }),
    setPasswordEditingState: (state) => ({
        type:    types.SET_PASSWORD_EDITING_STATE,
        payload: state,
    }),
});
