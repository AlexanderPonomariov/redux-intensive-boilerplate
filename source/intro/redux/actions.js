// Instruments
import { CHANGE_PHOTO } from './types';

export const changePhoto = (photo) => ({
    type:    CHANGE_PHOTO,
    payload: photo,
});
