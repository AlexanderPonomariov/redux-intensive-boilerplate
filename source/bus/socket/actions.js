// Instruments
import { socket } from '../../init/socket';
import { uiActions } from '../ui/actions';
import { postsActions } from '../posts/actions';

export const socketActions = Object.freeze({
    listenConnection: () => (dispatch) => {
        socket.on('connect', () => {
            dispatch(uiActions.setOnlineState(true));
        });

        socket.on('disconnect', () => {
            dispatch(uiActions.setOnlineState(false));
        });
    },

    listenPosts: () => (dispatch, getState) => {
        socket.on('create', (response) => {
            const { data, meta } = JSON.parse(response);

            if (getState().profile.get('id') === meta.userID) {
                return;
            }

            dispatch(postsActions.createPost(data));
        });

        socket.on('remove', (response) => {
            const { data, meta } = JSON.parse(response);

            if (getState().profile.get('id') === meta.userID) {
                return;
            }

            dispatch(postsActions.removePost(data));
        });

        socket.on('like', (response) => {
            const { data, meta } = JSON.parse(response);

            if (getState().profile.get('id') === meta.userID) {
                return;
            }

            if (meta.action === 'like') {
                const liker = getState()
                    .users.find((user) => user.get('id') === data.userID)
                    .delete('avatar');

                dispatch(
                    postsActions.likePost({
                        postID: data.postID,
                        liker,
                    })
                );

                return;
            }

            dispatch(postsActions.unlikePost(data));
        });
    },
});
