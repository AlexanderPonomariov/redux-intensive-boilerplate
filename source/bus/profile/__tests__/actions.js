// Core
import { profileActions } from '../actions';

describe('profile actions:', () => {
    test('fillProfile snapshot', () => {
        expect(profileActions.fillProfile(__.userProfile)).toMatchSnapshot();
    });

    test('updateAvatar snapshot', () => {
        expect(profileActions.updateAvatar(__.url)).toMatchSnapshot();
    });

    test('clearProfile snapshot', () => {
        expect(profileActions.clearProfile()).toMatchSnapshot();
    });
});
