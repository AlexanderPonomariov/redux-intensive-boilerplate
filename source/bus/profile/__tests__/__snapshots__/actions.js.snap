// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile actions: clearProfile snapshot 1`] = `
Object {
  "type": "CLEAR_PROFILE",
}
`;

exports[`profile actions: fillProfile snapshot 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "Walter",
    "id": "TEST_ID",
    "lastName": "White",
    "token": "TEST_TOKEN",
  },
  "type": "FILL_PROFILE",
}
`;

exports[`profile actions: updateAvatar snapshot 1`] = `
Object {
  "payload": "https://www.url.com",
  "type": "UPDATE_AVATAR",
}
`;
